lib/queue_classic_plus/worker.rb

```diff
@@ -19,7 +19,13 @@ def handle_failure(job, e)
       # not have kicked in yet and we might be in a failed transaction. To be
       # *absolutely* sure the retry/failure gets enqueued, we do a rollback just
       # in case (and if we're not in a transaction it will be a no-op).
-      QC.default_conn_adapter.execute 'ROLLBACK'
+      begin
+        QC.default_conn_adapter.execute 'ROLLBACK'
+      rescue PG::UnableToSend => e
+        # Using a new connection because the default connection was killed
+        QueueClassicPlus.logger.info "Creating new connection for job #{job[:id]}"
+        QC.default_conn_adapter = QC::ConnAdapter.new
```

Comment:
I think you need something akin to this: https://github.com/rainforestapp/Rainforest/blob/develop/config/unicorn.rb#L25 (since we're using ActiveRecord under the hood).